### PR TITLE
feedbackmodal1

### PR DESCRIPTION
**Pull request by Goktugberke**

### CODE REVIEW

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+          alt=""  className={this.decorateCSS("icon")}/>
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:181]: image alt field should not be empty

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    }
+  }
+
+  getName(): string {
+    return "FeedbackModal1";
+  }
+
+  render() {
+    const inputItems = this.castToObject<any[]>("input_items");
+    const placeholder = inputItems?.[0]?.value?.[1]?.value?.[0]?.value?.[0]?.value || "Add a Comment...";
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:175]: do not access values like this

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+      key: "headerTitle",
+      displayer: "Header Title",
+      value: "How are you feeling?",
+    });
+
+    this.addProp({
+        type:"icon",
+        key:" feedbackIcon",
+        displayer:"Feedback Icon",
+        value:"https://www.google.com/imgres?q=comment%20icon&imgurl=https%3A%2F%2Ficons.veryicon.com%2Fpng%2Fo%2Fhardware%2Fjackdizhu_pc%2Fcomment-25.png&imgrefurl=https%3A%2F%2Fwww.veryicon.com%2Ficons%2Fhardware%2Fjackdizhu_pc%2Fcomment-25.html&docid=fMJtLYW6rlY7AM&tbnid=iadJ3A-hOcv3EM&vet=12ahUKEwjJlcWx8pmHAxXrQ_EDHaERB_cQM3oECBkQAA..i&w=512&h=512&hcb=2&ved=2ahUKEwjJlcWx8pmHAxXrQ_EDHaERB_cQM3oECBkQAA"
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:37]: this should not be a link. it should be name of the icon. e.g. : FaArrowDown

[please read](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md)

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+  color: string;
+};
+
+const emojis: Emoji[] = [
+  { id: 1, label: 'Very Bad', icon: '😢', color: '#E74C3C' },
+  { id: 2, label: 'Bad', icon: '😞', color: '#F39C12' },
+  { id: 3, label: 'Medium', icon: '😐', color: '#F1C40F' },
+  { id: 4, label: 'Good', icon: '😊', color: '#2ECC71' },
+  { id: 5, label: 'Very Good', icon: '😂', color: '#3498DB' },
+];
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:20]: icon cannot be an emoji. also remove this array. add this prop hard coded into the constructor. remove the map.

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+          ]
+        }
+      ]
+    });
+
+    this.addProp({
+      type: "array",
+      key: "emojiProps",
+      displayer: "Emojis",
+      value: emojis.map(emoji => ({
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:118]: this should be hard coded. remove map

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 6]
```diff
@@ -0,0 +1,249 @@
+import * as React from "react";
+import { BaseModal } from "../../EditorComponent";
+import styles from "./feedbackmodal1.module.scss";
+import { ErrorMessage, Formik, Form, Field } from "formik";
+import * as Yup from "yup";
+
+type Emoji = {
+  id: number;
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:8]: revisit type according to feedbacks

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 7]
```diff
@@ -0,0 +1,249 @@
+import * as React from "react";
+import { BaseModal } from "../../EditorComponent";
+import styles from "./feedbackmodal1.module.scss";
+import { ErrorMessage, Formik, Form, Field } from "formik";
+import * as Yup from "yup";
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:5]: never import all from a package. only import things that you need.

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
...5GG7R+1tX/knRcwhGH+wBCG+QNDGOYPDGGYPzEMJ5thXDZV/vQRw1D/o8PtST0/bP8cH4YwhCEMYQhDGP6GMc16MorLpMq/zDC3gGH+wBCG+QNDGOYPDGGYPzCsYzh+ik2VY2QwP6xj2P45PgxhCEMYwhCGMPwdY5qXuCQ3zAsYwjB/YAjD/IEhDPMHhjDMn19t2JLTPT8VC716gGrYyhNaW3LKrlqCRD9l90Np2yaJ7gaU6nnPeqESst4i9bRrcuy6/lsjJ5ZLqU2iox4jbGQEoi/1MSVVdfRT50lXIzQjqtnQ6GhoR9v9ShBfY2gloJHRvva05w8tamjWJitpaSh5nQ15RrvvzBWkI5L3U9SK/nF3SK+GVLF0aWK0wl18SSv9NVLP6G+zZXAOuj1FV0fIELxY6IHbKgG9GYrPi3GVI35SMFgzL/is+6i4IrLfm/Ph8crBWSFSY69+Mgav2ykTsiPXlQF1gCv8heXT/21RMIczV1b+jwcbro7+L4vC0dlD+guRhhoaNQnvhmfu5y0mG2bYT1aW/cW7W9N3F4MM01UVPgbM3vvusvEBhuXW+Q0x2QaNUgrnHhm/4T5dYe/gwqJjRzfoNYy/h8jGe5UN0/Nv29d4DGcBKTUOp6oTvjdLb+g2nPAfis+5zkL9asT1GC7DFTuGis/rvMpYklDMnkaPSxK3w9CsCf3DNCLD5eduvWo+y1sFGQ7eOb+liEnmLMSQHwxV2XaXkADDgt2Fuan987gzfsMxmyPkFPb2GhqrCD98Cap77TGcnTjBhYgu5h9uwxHnJ2y51WXYZ5P8q6Qb2HEa8hNBca8ErIYltzjXHcp7hWwz3LM5QsgyMsFiyCb5k5QkT2ANL+xih74BQgicYY/z+76kDbQ2pmGfTfKH1IHWxjBkJ4LfcrfD6YYDdrHwnPtLKhfEcLXn/KRMBC0QQ3aBaiMvyRNmnJSKxL0pBI/hQ5VT8fLEbSh3N+oVl+FR1kTQgsPwIDlHXLEbipsIWrAZCpwIWrAYSpwIWmANp4IWC71whkInghZMw6P8JE8wDB+FLRZ60Q0FTwQtUMNtO5I8gRjKnghaUAzFvBGsxtVQ/ETQwn/DjzYlecI/wzZMBC38NXxuX4648mN4bluSJ8zaMxG0cGlpjrjS6gcUAAAAAAAAAMBf/gDj4HfRpmJ3XwAAAABJRU5ErkJggg=="
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:180]: this should be a prop. also add this image using a google storage link

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+        </div>
+        <div className={this.decorateCSS("headerContent")}>
+          <h2>{this.getPropValue("headerTitle")}</h2>
+        </div>
+        <div className={this.decorateCSS("descriptionContent")}>
+          <p>{this.getPropValue("description")}</p>
+        </div>
+        <div className={this.decorateCSS("feedbackModalEmojis")}>
+          {this.castToObject<any[]>("emojiProps")?.map((item: any, index: number) => {
+            const label = item?.value?.[0]?.value;
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:193]: do not access values like this

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+          <button className={this.decorateCSS("feedbackModalCloseButton")}>&times;</button>
+        </div>
+        <div className={this.decorateCSS("headerContent")}>
+          <h2>{this.getPropValue("headerTitle")}</h2>
+        </div>
+        <div className={this.decorateCSS("descriptionContent")}>
+          <p>{this.getPropValue("description")}</p>
+        </div>
+        <div className={this.decorateCSS("feedbackModalEmojis")}>
+          {this.castToObject<any[]>("emojiProps")?.map((item: any, index: number) => {
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:192]: do not use any. write a reliable type for this. also cast it while mapping too.
`.map((item: TYPE_THAT_YOUVE_WRITTEN, index....`

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+            const color = item?.value?.[2]?.value;
+
+            return (
+              <span
+                key={index}
+                id={`emoji-${label}`}
+                className={this.decorateCSS("feedbackModalEmoji")}
+                onClick={() => this.handleEmojiClick(label, color)}
+              >
+                <span className={this.decorateCSS("feedbackModalEmojiIcon")}>{icon}</span>
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:204]: do not render relevant elements if there is no icon entered. do this operation for all props

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+    background: linear-gradient(to bottom right, #418d7c, #28a745);
+    color: #fff;
+    font-size: 1rem;
+    cursor: pointer;
+    transition: background 0.3s;
+
+    &:hover {
+      background: linear-gradient(to bottom right, #418d7c, #218838);
+    }
+  }
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:195]: add media queries for mobile

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
handle the changes

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+
+    return (
+      <div className={this.decorateCSS("feedbackModal")}>
+        <div className={this.decorateCSS("feedbackModalHeader")}>
+          <ComposerIcon
+            name={this.getPropValue("feedback_icon")}
+            propsIcon={{
+              className: `${this.decorateCSS("icon")}`,
+            }}
+          />
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:199]: if feedback_icon prop is empty, don't render composerIcon

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 14]
```diff
@@ -0,0 +1,268 @@
...
+          <h1 className={this.decorateCSS("feedbackModalHeaderh1")}>
+            {this.getPropValue("title")}
+          </h1>
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:202]: if title prop is empty, don't render this h1 element.
[read here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 15]
```diff
@@ -0,0 +1,268 @@
...
+          <button className={this.decorateCSS("feedbackModalCloseButton")}>&times;</button>
+        </div>
+        <div className={this.decorateCSS("headerContent")}>
+          <h2>{this.getPropValue("headerTitle")}</h2>
+        </div>
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:207]: if title prop is empty, don't render this div element.
[read here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 16]
```diff
@@ -0,0 +1,268 @@
...
+        <div className={this.decorateCSS("descriptionContent")}>
+          <p>{this.getPropValue("description")}</p>
+        </div>
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:210]: if title prop is empty, don't render this div element.
[read here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 17]
```diff
@@ -0,0 +1,268 @@
...
+              <ComposerIcon
+              
+              name={this.getPropValue("emoji_icon")}
+                propsIcon={{
+                  className: `${this.decorateCSS("feedbackModalEmojiIcon")}`,
+                }}
+              />
+            </span>
+          ))}
+        </div>
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:228]: render this div only if emojiProps.length > 0

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 18]
```diff
@@ -0,0 +1,268 @@
...
+                  className={this.decorateCSS("input")}
+                  rows={4}
+                />
+                <ErrorMessage
+                  className={this.decorateCSS("error-message")}
+                  name="message"
+                  component={"span"}
+                />
+                <button type="submit" className={this.decorateCSS("feedbackModalSubmit")}>
+                  Submit Now
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:257]: transform this "Submit Now" text into a prop. user should be able to change it

**Review by dogusmiuw — CHANGES_REQUESTED** [review]

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 19]
```diff
@@ -0,0 +1,295 @@
+.feedbackModal {
+  width: 400px;
+  padding: 20px;
+  border-radius: 10px;
+  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+  background-color: var(--composer-html-background);
+  font-family: 'Arial', sans-serif;
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:7]: do not use font family

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 20]
```diff
@@ -0,0 +1,295 @@
...
+  transition: background-color 0.3s ease;
+
+  .feedbackModalHeader {
+    display: flex;
+    align-items: center;
+    border-bottom: 1px solid #eee;
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:13]: do not use static color

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 21]
```diff
@@ -0,0 +1,295 @@
...
+    .feedbackModalEmoji {
+      position: relative;
+      cursor: pointer;
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+      padding: 10px;
+      border-radius: 50%;
+      transition: background-color 0.3s, transform 0.3s;
+      background-color: #d2d1d13a;
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:90]: do not use static colors

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 22]
```diff
@@ -0,0 +1,295 @@
...
+      font-size: 1rem;
+      color:  var(--composer-font-color-primary);
+      margin-bottom: 20px;
+    }
+  }
+
+  .feedbackModalEmojis {
+    display: flex;
+    justify-content: space-around;
+    flex-direction: row;
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:77]: default value is already : row. remove this line

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 23]
```diff
@@ -0,0 +1,295 @@
...
+    margin-bottom: 20px;
+    flex-wrap: wrap;
+  
+    .feedbackModalEmoji {
+      position: relative;
+      cursor: pointer;
+      display: flex;
+      flex-direction: row;
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:85]: default value is already : row. remove this line

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 24]
```diff
@@ -0,0 +1,295 @@
...
+    justify-content: center;
+    
+  
+      .input {
+        position: relative;
+        width: 360px;
+        height: 95px;
+        border-color: var(--composer-font-color-primary);
+        border-radius:  var(--composer-border-radius);;
+        //border: 0.5px solid rgba(var(--composer-font-color-primary-rgb), 0.3);
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:163]: remove comment line

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 25]
```diff
@@ -0,0 +1,295 @@
...
+      }
+  
+      
+    
+  }
+
+  .feedbackModalComment {
+    width: 100%;
+    height: 60px;
+    border: 1px solid #ddd;
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:181]: do not use static colors

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 26]
```diff
@@ -0,0 +1,295 @@
...
+    background: var(--composer-secondary-color);
+    color: var(--composer-font-color-secondary);
+    font-size: 1rem;
+    cursor: pointer;
+    transition: background 0.3s;
+
+    
+  }
+
+  @media (max-width: 768px) {
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:202]:
```suggestion
  @media only screen and (max-width: 768px) { 
```

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss [hunk 27]
```diff
@@ -0,0 +1,295 @@
...
+      padding: 15px 0;
+    }
+
+    .feedbackModalSubmit {
+      font-size: 0.875rem;
+      padding: 8px;
+    }
+  }
+
+  @media (max-width: 480px) {
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.module.scss:249]:
```suggestion
  @media only screen and (max-width: 480px) { 
```

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 28]
```diff
@@ -0,0 +1,212 @@
...
+import styles from "./feedbackmodal1.module.scss";
+import { object, string } from "yup";
+import { ComposerIcon } from "../../../composer-base-components/icon/icon";
+
+type Emoji = {
+  type: "icon";
+  key: string;
+  displayer: string;
+  value: string;
+};
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:12]: invalid type.
[please read](https://github.com/Teknodev/composer-tools/tree/main/editor-components#how-to-approach-props)

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 29]
```diff
@@ -0,0 +1,212 @@
...
+              name={this.getPropValue("close_icon")}
+              propsIcon={{
+                className: `${this.decorateCSS("close_icon")}`,
+              }}
+            />
+          </button>
+        </div>
+
+        <div className={this.decorateCSS("headerContent")}>
+          {isHeaderTitleExist &&
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:159]: do not just remove h2. remove all the div

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 30]
```diff
@@ -0,0 +1,212 @@
...
+            <h2>{this.getPropValue("headerTitle")}</h2>
+          }
+        </div>
+
+        <div className={this.decorateCSS("descriptionContent")}>
+          {isDescriptionExist &&
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:165]: do not just remove p, remove all the div

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 31]
```diff
@@ -0,0 +1,212 @@
...
+            <p>{this.getPropValue("description")}</p>
+          }
+        </div>
+        
+        {emojiProps.length > 0 && (
+          <div className={this.decorateCSS("feedbackModalEmojis")}>
+            {emojiProps.map((item: Emoji, index: number) => (
```
Comment (dogusmiuw) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:172]:
```suggestion
            {emojiProps.map((item: Emoji, index: number) => {
            
            if(item.value)
                return (
```

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle the changes

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 32]
```diff
@@ -220,6 +222,10 @@ class FeedbackModal1 extends BaseModal {
     }
   }
 
+  validationSchema = Yup.object().shape({
+    message: Yup.string().min(5, "Min 5 character!").required("Required"),
```
Comment (Ayazelar) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:226]: the messages must come from the prop. user may want to change message

File: editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx [hunk 33]
```diff
@@ -322,25 +328,43 @@ class FeedbackModal1 extends BaseModal {
...
-                <button className={this.decorateCSS("button")}>
-                  {buttonval}
-                </button>
+        {(buttonval || inputPlaceholder) &&
+          (<div className={this.decorateCSS("contact-form")}>
+            <Formik
+              initialValues={{ message: "" }}
+              validationSchema={this.validationSchema}
+              onSubmit={(data, { resetForm }) => {
+                this.insertForm("Contact Us", data);
```
Comment (Ayazelar) [inline editor-components/feedback/feedbackmodal1/feedbackmodal1.tsx:337]: data should be like this= {
input_0: "",
input_1: ""
}

**Review by Ayazelar — CHANGES_REQUESTED** [review]
change

**Review by Ayazelar — CHANGES_REQUESTED** [review]
inserted form data should be like this= {
input_0: "Message of the input. lOrem ips",
input_1: ""
}

**Pull request closed without merging**